### PR TITLE
Develop to master

### PR DESCRIPTION
**Pull request by eugene-chr**

### CODE REVIEW

File: components/shared/Dropdown/container.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+const Dropdown = ({ title, children, variant }) => {
+  const [isShowDropdown, setShowDropdown] = useState(false);
+  const handleHideDropdown = () => {
+    setShowDropdown(false);
+    // document.body.removeEventListener('click', handleHideDropdown);
+  };
+  const handleToggleDropdown = () => {
+    // document.body.addEventListener('click', handleHideDropdown, true);
+    setShowDropdown(!isShowDropdown);
```
Comment (plzen) [inline components/shared/Dropdown/container.js:13]: It's better to update state with callback style, like
```
setShowDropdown(prevValue => !!prevValue)
```

File: components/shared/MediaCard/container.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { MediaCardComponent } from './component';
+
+const MediaCard = ({ href, asPath, variant, children, vote_average, image, title, small }) => {
+  const [isShow, setIsShow] = useState(false);
+
+  const handleToggleInfo = () => {
+    setIsShow(!isShow);
```
Comment (plzen) [inline components/shared/MediaCard/container.js:9]: Use callback style

File: components/shared/Pagination/component.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        });
+        break;
+      default:
+        break;
+    }
+
+    return acc;
+  }, []);
+
+  const asPath = () => {
```
Comment (plzen) [inline components/shared/Pagination/component.js:50]: Move this business logic to container

File: components/shared/Pagination/component.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    if (pathname.includes('people')) {
+      return '/people';
```
Comment (plzen) [inline components/shared/Pagination/component.js:52]: Pass path from parent component and avoid if statement here

File: components/shared/Sidebar/component.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+import { MAIN_BUTTONS_OFFSET, Z_INDEX } from 'constants/layout';
+import { SidebarContext } from 'components/shared/Sidebar/context';
+import { Menu } from 'components/shared/Menu';
+import { CloseIcon } from 'components/shared/svgIcons/CloseIcon';
+import { Button } from 'components/shared/Button';
+import { sideBarSlideInRight } from 'utils/animations';
+
+const Sidebar = () => {
+  const { handleHideSidebar } = useContext(SidebarContext);
```
Comment (plzen) [inline components/shared/Sidebar/component.js:14]: Use container for hooks

File: pages/_app.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import { Layout } from 'components/Layout';
+import { messages_en } from 'locales/en';
+import { messages_ru } from 'locales/ru';
+import { messages_uk } from 'locales/uk';
+import { DEFAULT_LOCALE } from 'constants/layout';
+
+const cache = createIntlCache();
+
+const MyApp = ({ Component, pageProps }) => {
+  const cookies = new Cookies();
```
Comment (plzen) [inline pages/_app.js:20]: Use server-side cookies also

File: utils/getPerson.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { API_KEY, BASE_URL } from 'constants/tmdb';
+
+export const getPerson = (locale, personId) => `${BASE_URL}/person/${personId}?api_key=${API_KEY}&language=${locale}`;
```
Comment (plzen) [inline utils/getPerson.js:3]: Is it really needed? Can't it be implemented in concepts?

File: components/Homepage/component.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        </ButtonLink>
+      </ReadMore>
+    </LeftColumn>
+    <RightColumn key="RightColumn" variants={rightColumn} initial="initial" animate="animate" exit="exit">
+      <Poster path={poster_path} alt={title} size={4} />
+    </RightColumn>
+  </>
+);
+
+const LeftColumn = styled(motion.div)`
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/Homepage/component.js:33]: discussed

File: components/Layout/component.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+`;
+
+LayoutComponent.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
+  showSidebar: PropTypes.bool.isRequired,
+  handleShowSidebar: PropTypes.func.isRequired,
+  handleHideSidebar: PropTypes.func.isRequired,
+};
+
+export { LayoutComponent };
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/Layout/component.js:51]: discussed

File: components/Layout/container.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+      {children}
+    </LayoutComponent>
+  );
+};
+
+Layout.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
+};
+
+export { Layout };
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/Layout/container.js:28]: discussed

File: components/Movie/component.js [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+        </span>
+        <Title as="h2">
+          <FormattedMessage id="movie.originalLanguage" />
+        </Title>
+        <span>{original_language.toUpperCase()}</span>
+        <Title as="h2">
+          <FormattedMessage id="movie.duration" />
+        </Title>
+        <span>
+          <FormattedMessage id="units.hours">{(text) => `${formatTime(runtime).hours}${text} `}</FormattedMessage>
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/Movie/component.js:64]: done

File: components/Movie/component.js [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+        {inRange(vote_average, 0.1, 10) && <RatingCircle voteAverage={vote_average} />}
+        {homepage && (
+          <ButtonLink variant="primary" href={homepage} animate>
+            <FormattedMessage id="movie.official" />
+          </ButtonLink>
+        )}
+        {imdb_id && (
+          <ButtonLink variant="primary" href={`https://www.imdb.com/title/${imdb_id}`} animate>
+            <FormattedMessage id="movie.showOn" tagName="span" />
+            <img src="/images/imdb.svg" alt="IMDB Rating" style={{ width: '50px', marginLeft: '12px' }} />
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/Movie/component.js:86]: done

File: components/People/container.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { PeopleComponent } from 'components/People/component';
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/People/container.js:3]: done

File: components/TVShow/container.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { TVShowComponent } from './component';
+
+const TVShow = ({ tvShow }) => {
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/TVShow/container.js:6]: done

File: components/TVShows/component.js [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+            rating={vote_average}
+            image={poster_path}
+            title={name}
+            asPath={`/tv-shows/${id}`}
+          >
+            <Genres items={genresSelector(genres, genre_ids)} />
+          </MediaCard>
+        ))}
+      </MediaGrid>
+      <Pagination totalPages={totalPages} page={page} router={router} />
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/TVShows/component.js:30]: done

File: components/shared/Button/component.js [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+    <Container
+      className={className}
+      variant={variant}
+      variants={animate && buttonAnimation}
+      whileHover="whileHover"
+      whileTap="whileTap"
+      onClick={handleClick}
+      style={style}
+    >
+      {children && children}
```
[the remark this answers is not in the document]
Comment (eugene-chr) [reply components/shared/Button/component.js:23]: done

File: utils/getPerson.js (hunk 7 above)
Comment (eugene-chr) [reply utils/getPerson.js:3]: done

File: components/shared/Pagination/component.js (hunk 3 above)
Comment (eugene-chr) [reply components/shared/Pagination/component.js:50]: done

File: components/shared/Sidebar/component.js (hunk 5 above)
Comment (eugene-chr) [reply components/shared/Sidebar/component.js:14]: done

File: components/shared/Pagination/component.js (hunk 4 above)
Comment (eugene-chr) [reply components/shared/Pagination/component.js:52]: done

File: components/shared/Dropdown/container.js (hunk 1 above)
Comment (eugene-chr) [reply components/shared/Dropdown/container.js:13]: done

File: components/shared/MediaCard/container.js (hunk 2 above)
Comment (eugene-chr) [reply components/shared/MediaCard/container.js:9]: done

**Pull request merged**